src/tools/spectest-interp.cc

```diff
@@ -490,65 +497,131 @@ wabt::Result JSONParser::ParseConst(TypedValue* out_value) {
   PARSE_KEY_STRING_VALUE("value", &value_str);
   EXPECT("}");
 
+  return ParseConstValue(out_value, type_str, value_str);
+}
+
+wabt::Result JSONParser::ParseConstValue(TypedValue* out_value,
+                                         string_view type_str,
+                                         string_view value_str) {
   const char* value_start = value_str.data();
   const char* value_end = value_str.data() + value_str.size();
-
   if (type_str == "i32") {
     uint32_t value;
-    CHECK_RESULT(
-        ParseInt32(value_start, value_end, &value, ParseIntType::UnsignedOnly));
+    if (Failed((ParseInt32(value_start, value_end, &value,
```

Comment:
why change this?

Reply:
Oh.. these errors were not being reported.  This is a separate issue that I will split out.